zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/Job.java

```diff
@@ -256,4 +257,21 @@ private Logger logger() {
   protected void setResult(Object result) {
     this.result = result;
   }
+
+  public Job findSimilarJob(List<Job> queue) {
+    for (int i = 0; i < queue.size(); i++) {
+      Job j = queue.get(i);
+      if (j.getId().equals(this.getId())) {
+        return j;
+      }
+    }
+    return null;
+  }
+
+  public boolean isSimilarJob(Job j) {
```

Comment:
Also check for j being null and throw NullArgumentException with the proper arg name.